client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -44,16 +45,24 @@ function getChecklistCompletionProgress( tasks: Task[] | null ) {
 const Sidebar = ( { siteSlug, submit, goNext, goToStep }: SidebarProps ) => {
 	let siteName = '';
 	let topLevelDomain = '';
+	const [ showLaunchTitle, setShowLaunchTitle ] = useState< boolean >( false );
 	const flow = useFlowParam();
 	const translate = useTranslate();
 	const site = useSite();
-	const translatedStrings = getLaunchpadTranslations( flow );
+	const { flowName, title, launchTitle, subtitle } = getLaunchpadTranslations( flow );
 	const arrayOfFilteredTasks: Task[] | null = getArrayOfFilteredTasks( tasks, flow );
 	const enhancedTasks =
 		site && getEnhancedTasks( arrayOfFilteredTasks, siteSlug, site, submit, goToStep );
 
 	const taskCompletionProgress = site && getChecklistCompletionProgress( enhancedTasks );
 
+	useEffect( () => {
```

Comment:
I'm not sure why we're using an effect here. Is it to ensure that this these methods are only called once?

Reply:
@jeyip - I added the useEffect because I'm calling setShowLaunchTitle(). My understanding is that if you call that kind of setState call directly in the root of a component (ie, no useEffect or event handler, etc), it'll cause a re-rendering loop (render which updates state which causes rerender which updates state, etc). I just tested that to confirm, and that does seem to be what I'm seeing.

Reply:
I might be missing something here, but I guess I'm confused as to why we use setState at all 🤔 

```
const launchTask = enhancedTasks?.find( ( task ) => task.isLaunchTask === true );
const showLaunchTitle = launchTask && ! isTaskDisabled( launchTask )
```

Reply:
> I'm confused as to why we use setState at all

And you are totally correct. I guess I'm overcomplicating things here. We don't need the state here, and that means don't need the useEffect, and the net result is a lot simpler. Just pushed the changes.